File: frontend/src/types/dto/response/LobbyDetailsDto.ts
import type { Difficulty } from "@/types/enum/Difficulty"
import type { GameMode } from "@/types/enum/GameMode"
import type { GameType } from "@/types/enum/GameType"
import type { TimeLimitPreset } from "@/types/enum/TimeLimitPreset"

export type LobbyDetailsDto = {
    id: number,
    lobbyName: string,
    isActive: boolean,
    inGame: boolean,
    currentGameId: number | null,
    gameType: GameType,
    gameMode: GameMode,
    difficulty: Difficulty,
    timeLimitPreset: TimeLimitPreset
}

File: backend/src/main/java/com/github/ryand6/sudokuweb/dto/response/LobbyDetailsDto.java
package com.github.ryand6.sudokuweb.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.github.ryand6.sudokuweb.enums.Difficulty;
import com.github.ryand6.sudokuweb.enums.GameMode;
import com.github.ryand6.sudokuweb.enums.GameType;
import com.github.ryand6.sudokuweb.enums.TimeLimitPreset;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class LobbyDetailsDto {

    private Long id;

    private String lobbyName;

    @JsonProperty("isActive")
    private boolean isActive;

    private boolean inGame;

    private Long currentGameId;

    private GameType gameType;

    private GameMode gameMode;

    private Difficulty difficulty;

    private TimeLimitPreset timeLimitPreset;

}
